deploy/cdk/lib/user-content/index.ts

```diff
@@ -0,0 +1,119 @@
+import apigateway = require('@aws-cdk/aws-apigateway');
+import dynamodb = require('@aws-cdk/aws-dynamodb');
+import lambda = require('@aws-cdk/aws-lambda');
+import ssm = require('@aws-cdk/aws-ssm');
+import cdk = require('@aws-cdk/core');
+
+export interface UserContentStackProps extends cdk.StackProps {
+  readonly lambdaCodePath: string
+  readonly tokenAudiencePath: string
+  readonly tokenIssuerPath: string
+};
+
+export class UserContentStack extends cdk.Stack {
+  constructor(scope: cdk.Construct, id: string, props: UserContentStackProps) {
+    super(scope, id, props);
+
+    // Dynamo Tables
+    const userDynamoTable = new dynamodb.Table(this, 'UsersTable', {
+      billingMode: dynamodb.BillingMode.PAY_PER_REQUEST,
+      partitionKey: {
+        name: 'userName',
+        type: dynamodb.AttributeType.STRING
+      },
+      tableName: `${this.stackName}-users`,
+      removalPolicy: cdk.RemovalPolicy.DESTROY, // NOT recommended for production code
+    });
+
+    const collectionDynamoTable = new dynamodb.Table(this, 'CollectionsTable', {
+      billingMode: dynamodb.BillingMode.PAY_PER_REQUEST,
+      partitionKey: {
+        name: 'uuid',
+        type: dynamodb.AttributeType.STRING
+      },
+      tableName: `${this.stackName}-collections`,
+      removalPolicy: cdk.RemovalPolicy.DESTROY, // NOT recommended for production code
+    });
+    collectionDynamoTable.addGlobalSecondaryIndex({
+      indexName: 'userName',
+      partitionKey: {
+        name: 'userName',
+        type: dynamodb.AttributeType.STRING
+      }
+    });
+
+    const itemDynamoTable = new dynamodb.Table(this, 'ItemsTable', {
```

Comment:
I wonder if these shouldn't just be a single table, with items/collections as properties of a user. Maybe read through https://docs.aws.amazon.com/amazondynamodb/latest/developerguide/bp-general-nosql-design.html.

Reply:
Quite possibly. Jon, Michael, and I discussed a few different approaches and agreed that for the first iteration having three table would be the simplest approach. 

There is still a lot to discuss with the product owner as to how the user content features will actually work. I honestly, expect a lot of the requirements around this part of the application to change possibly fairly dramatically, so I've kind of opted to have something simple vs something that is optimized around a data structure that will likely change.